lib/problems.rb

```diff
@@ -1,9 +1,38 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) - length of the string
+# Space Complexity: O(n) - # of opening brackets
+
+# Given a string containing opening and closing braces, check if it represents a balanced expression or not.
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
+  stack = Stack.new
+  opening = ["(", "[", "{"]
+
+  string.each_char do |char|
+    # if character is an opening bracket
+    if opening.include?(char)
+      # push to stack
+      stack.push(char)
+    # if not
+    else
+      # pop the top character from the stack
+      top_stack = stack.pop
+      if char == ')'
+        return false if top_stack != '('
+      elsif char == ']'
+        return false if top_stack != '['
+      elsif char == '}'
+        return false if top_stack != '{'
+        # if the current character is a closing bracket
+          # return false if the top stack character is not a matching open
+      end
```

Comment:
Instead of this, you could simplify it with a hash.  where the key is the close brace and the matching open brace is the value.